engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -256,20 +255,30 @@ class WorkflowActor(workflowId: WorkflowId,
 
   onTransition {
     case oldState -> terminalState if terminalState.terminal =>
-      log.info(s"$tag transition from $oldState to $terminalState: shutting down")
+      workflowLogger.info(s"transition from $oldState to $terminalState: shutting down")
       // Add the end time of the workflow in the MetadataService
       val now = OffsetDateTime.now
-      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime),
-        MetadataValue(now))
+      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime), MetadataValue(now))
       serviceRegistryActor ! PutMetadataAction(metadataEventMsg)
       terminalState match {
         case WorkflowSucceededState =>
           context.parent ! WorkflowSucceededResponse(workflowId)
         case WorkflowFailedState => // The failure will already have been sent. So just kick back and relax here.
         case WorkflowAbortedState =>
           context.parent ! WorkflowAbortedResponse(workflowId)
-        case unknownState => log.warning(s"$tag $unknownState is an unhandled terminal state!")
+        case unknownState => workflowLogger.warn(s"$unknownState is an unhandled terminal state!")
       }
+
+      // Copy/Delete workflow logs
+      if (WorkflowLogger.isEnabled) {
+        stateData.workflowDescriptor foreach { wd =>
+          // Create the actor in the system context because this WorkflowActor is going to die right after
+          val copyWorkflowLogsActor: ActorRef = context.system.actorOf(CopyWorkflowLogsActor.props(wd), s"CopyWorkflowLogsActor-$workflowId")
+          copyWorkflowLogsActor ! CopyWorkflowLogsActor.Start
```

Comment:
tl;dr feel free to do it how you want

FWIW I also am not a huge fan of what @kshakir is talking about, although more just because it seems superfluous vs his more reasoned approach. I don't really care enough to tell people to do it otherwise. 

I think that tactic started due to some of the timing dances we needed to do. If I'm not just making that statement up that seems like a bit of a smell that the timing dances are overly complicated